File: stack/MinStack.go
type MinStack struct {
	Stack []int
	Min   []int
}

func Constructor() MinStack {
	return MinStack{Stack: []int{}, Min: []int{}}

}
func min(x, y int) int {
	if x < y {
		return x
	}
	return y
}
func (this *MinStack) Push(val int) {
	this.Stack = append(this.Stack, val)
	if len(this.Min) != 0 {
		this.Min = append(this.Min, min(val, this.GetMin()))
	} else {
		this.Min = append(this.Min, val)

	}

}

func (this *MinStack) Pop() {
	this.Stack = this.Stack[:len(this.Stack)-1]
	this.Min = this.Min[:len(this.Min)-1]

}

func (this *MinStack) Top() int {
	return this.Stack[len(this.Stack)-1]
}

func (this *MinStack) GetMin() int {
	return this.Min[len(this.Min)-1]
}

File: stack/MinStack.py
class MinStack(object):

    def __init__(self):
        self.stack = []
        self.MinStack = []
        

    def push(self, val):
        """
        :type val: int
        :rtype: None
        """
        self.stack.append(val)
        minVal = min(val, self.MinStack[-1] if self.MinStack else val)
        self.MinStack.append(minVal)
        

    def pop(self):
        """
        :rtype: None
        """
        self.stack.pop()
        self.MinStack.pop()
        

    def top(self):
        """
        :rtype: int
        """
        return self.stack[-1]
        

    def getMin(self):
        """
        :rtype: int
        """
        return self.MinStack[-1]

        


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(val)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()